libs/core/src/lib/utils/drag-and-drop/dnd-item/dnd-item.directive.ts

```diff
@@ -0,0 +1,216 @@
+
+    constructor(public element: ElementRef, private _dragDrop: DragDrop) {}
+
+    /** @hidden */
+    public getElementChord(isBefore: boolean, listMode: boolean): ElementChord {
+        /** Takes distance from the beginning of window page */
+        const rect = <DOMRect>this.element.nativeElement.getBoundingClientRect();
+
+        const position: LinkPosition = isBefore ? 'before' : 'after';
+
+        /** Depending on the position, gets the left or right side of element */
+        const x = rect.left + (isBefore || listMode ? 0 : this.element.nativeElement.offsetWidth);
+
+        /** Vertically distance is counted by distance from top of the side + half of the element height */
+        return {
+            x: x,
+            position: position,
+            y: rect.top + this.element.nativeElement.offsetHeight / 2,
+            stickToPosition: this.stickInPlace
+        };
+    }
+
+    /** @hidden */
+    public ngAfterContentInit(): void {
+        if (!this.dndDisabled) {
+            this._setCDKDrag();
+        }
+    }
+
+    /** @hidden */
+    public onCdkMove(cdkMovedEvent: CdkDragMove): void {
+        this.moved.emit(cdkMovedEvent);
+    }
+
+    /** @hidden */
+    public onCdkDragReleased(): void {
+        /** Remove class which is added, when element is dragged */
+        this.element.nativeElement.classList.remove(this.CLASS_WHEN_ELEMENT_DRAGGED);
+        this.released.emit();
+
+        /** Resets the position of element. */
+        this._dragRef.reset();
+
+        /** Removes placeholder element */
+        this.removePlaceholder();
+    }
+
+    /** @hidden */
+    public onCdkDragStart(): void {
+        /** Adds class */
+        this.element.nativeElement.classList.add(this.CLASS_WHEN_ELEMENT_DRAGGED);
+        if (!this._placeholderElement) {
+            this.createPlaceHolder();
+        }
+        this.started.emit();
+    }
+
+    /** @hidden */
+    public removePlaceholder(): void {
+        if (this._placeholderElement && this._placeholderElement.parentNode) {
+            // IE11 workaround
+            this._placeholderElement.parentNode.removeChild(this._placeholderElement);
+            this._placeholderElement = null;
+        }
+    }
+
+    /** @hidden */
+    public removeLine(): void {
+        if (this._lineElement && this._lineElement.parentNode) {
+            // IE11 workaround
+            this._lineElement.parentNode.removeChild(this._lineElement);
+            this._lineElement = null;
+        }
+    }
+
+    /** @hidden */
+    public removeReplacement(): void {
+        if (this._replaceIndicator && this._replaceIndicator.parentNode) {
+            // IE11 workaround
+            this._replaceIndicator.parentNode.removeChild(this._replaceIndicator);
+            this._replaceIndicator = null;
+        }
+    }
+
+    /** @hidden */
+    public createReplaceIndicator(): void {
+        this._replaceIndicator = document.createElement('DIV');
```

Comment:
Why upper case `DIV`?

Reply:
I don't know, it is treated the same as a lowercase